### PR TITLE
Use ocaml.deprecated attribute on all stdlib deprecated functions.

### PR DESCRIPTION
**Pull request by jhjourdan**
Some of these function were deprecated before the attribute was created.

### CODE REVIEW

**Comment by alainfrisch** [comment]
Related : #2121.

**Comment by jhjourdan** [comment]
> Related : #2121.

Ah, indeed. I was not aware that someone else has done this work. Let's try to actually get this merged.

**Comment by jhjourdan** [comment]
Alright, I have just rebased #2121 and added a few bits from my side (mainly updating the boxedint test to use the non-deprecated `Printf.printf` instead).

File: stdlib/stdlib.mli [hunk 1]
```diff
@@ -1359,7 +1359,7 @@ module Oo           = Oo
 module Option       = Option
 module Parsing      = Parsing
 module Pervasives   = Pervasives
-[@@deprecated "Use Stdlib instead.\n\
+[@@deprecated "use the Stdlib module\n\
```
Comment (hannesm) [inline stdlib/stdlib.mli]: this is the only use of `[@@deprecated` in the codebase (all other uses are `[@@ocaml.deprecated`) -- is this intentional?

Comment (jhjourdan) [reply stdlib/stdlib.mli]: Fixed, thanks.

**Comment by jhjourdan** [comment]
The only CI failure is because th Changes file has not changed.

**Comment by damiendoligez** [comment]
At the last developer meeting, we decided to hold off deprecating anything before we settle on a deprecation workflow, so don't merge this PR too fast, thanks.

**Pull request closed without merging**